crypto/fipsmodule/ec/internal.h

```diff
@@ -745,6 +745,48 @@ const EC_METHOD *EC_GFp_nistp521_method(void);
 // x86-64 optimized P256. See http://eprint.iacr.org/2013/816.
 const EC_METHOD *EC_GFp_nistz256_method(void);
 
+
+// EC_KEY METHOD
+
+// ec_key_method_st is a structure of function pointers implementing EC_KEY
+// operations. Currently, AWS-LC only allows consumers to set certain fields.
+struct ec_key_method_st {
+    int (*init)(EC_KEY *key);
+    void (*finish)(EC_KEY *key);
+
+    // AWS-LC doesn't support custom values for EC_KEY operations
+    // as of now. |k_inv| and |r| must be NULL parameters.
+    // The |type| parameter is ignored in OpenSSL, we pass in zero for it
+    int (*sign)(int type, const uint8_t *digest, int digest_len,
+                uint8_t *sig, unsigned int *siglen, const BIGNUM *k_inv,
+                const BIGNUM *r, EC_KEY *eckey);
+
+    // AWS-LC doesn't support custom values for EC_KEY operations
+    // as of now. |k_inv| and |r| must be NULL parameters.
+    ECDSA_SIG *(*sign_sig)(const uint8_t *digest, int digest_len,
+                           const BIGNUM *in_kinv, const BIGNUM *in_r,
+                           EC_KEY *eckey);
+
+    // Currently, |EC_KEY_METHOD| only supports |ECDSA_FLAG_OPAQUE|. It is
+    // not set by default.
+    int flags;
+
+    // AWS-LC currently does not support these fields, must be NULL
+    int (*sign_setup)(EC_KEY *eckey, BN_CTX *ctx_in, BIGNUM **k_inv,
+                      BIGNUM **r);
+    int (*copy)(EC_KEY *dest, const EC_KEY *src);
+    int (*set_group)(EC_KEY *key, const EC_GROUP *group);
+    int (*set_private)(EC_KEY *key, const BIGNUM *priv_key);
+    int (*set_public)(EC_KEY *key, const EC_POINT *pub_key);
+    int (*keygen)(EC_KEY *key);
+    int (*compute_key)(unsigned char **out, size_t *out_len,
+                       const EC_POINT *pub_key, const EC_KEY *ecdh);
+    int (*verify)(int type, const uint8_t *digest, int digest_len,
+                  const uint8_t *sig, unsigned int sig_len, EC_KEY *eckey);
+    int (*verify_sig)(const uint8_t *digest, int digest_len,
+                      const ECDSA_SIG *sig, EC_KEY *eckey);
```

Comment:
In case we ever make the structs public to match OpenSSL and allow static allocation of the structs, it would be easier to match OpenSSL. We don't necessarily need the fields that we don't provide setters for yet. I have removed compute_key, keygen, verify, and verify_sig for now and updated documentation to reflect the above point.